indexer-balances/src/models/balance_changes.rs

```diff
@@ -45,6 +45,20 @@ impl crate::models::SqlxMethods for NearBalanceEvent {
             + " ON CONFLICT DO NOTHING")
     }
 
+    fn select_prev_balance_query(block_height: u64, account_id: &str) -> String {
+        format!(
+            "
+                 SELECT *
+                 FROM near_balance_events
+                 WHERE block_height < {}
+                 AND affected_account_id = '{}'
+                 ORDER BY event_index desc
+                 LIMIT 1;
+             ",
+            block_height, account_id
```

Comment:
Probably not a great idea to be injecting the arguments, should probably be using `.bind()` instead. But this allows returning arbitrary queries, allowing them to be used more generically.